pkg/services/opsgenie/opsgenie_json.go

```diff
@@ -0,0 +1,92 @@
+package opsgenie
+
+import (
+	"encoding/json"
+	"github.com/containrrr/shoutrrr/pkg/types"
+)
+
+// AlertPayload represents the payload being sent to the OpsGenie API
+//
+// See: https://docs.opsgenie.com/docs/alert-api#create-alert
+//
+// Some fields contain complex values like arrays and objects.
+// Because `params` are strings only we cannot pass in slices
+// or maps. Instead we "preserve" the JSON in those fields. That
+// way we can pass in complex types as JSON like so:
+//
+//	service.Send("An example alert message", &types.Params{
+//		"alias":       "Life is too short for no alias",
+//		"description": "Every alert needs a description",
+//		"responders":  `[{"id":"4513b7ea-3b91-438f-b7e4-e3e54af9147c","type":"team"},{"name":"NOC","type":"team"}]`,
+//		"visibleTo":   `[{"id":"4513b7ea-3b91-438f-b7e4-e3e54af9147c","type":"team"},{"name":"rocket_team","type":"team"}]`,
+//		"details":     `{"key1": "value1", "key2": "value2"}`,
+//	})
```

Comment:
Using raw JSON in the Query and params is not ideal. If the user know the API well and types JSON frequently it might work, but there is no feedback that something might not work.

Arrays are more or less universally supplied as `foo,bar,baz` in Shoutrrr services, which is easy to de-/serialize, but the others might need some more special treatment.

Perhaps responders/visibleTo could be supplied as 
`&responders=team:4513b7ea-3b91-438f-b7e4-e3e54af9147c,team:NOC` and `&visibleTo=team:4513b7ea-3b91-438f-b7e4-e3e54af9147c,team:rocket_team`?
And perhaps the details could just be: `details=key1:value1,key2:value2` ?

It's easier to manually input and since it's parsed we can actually give feedback to the user about the format and show an appropriate error when it's not in the correct format.
It will also be work better with `shoutrrr generate` and `shoutrr verify`.

With #92 you can get a lot of the de-/serialization of the params and query strings for free, so I guess this can be fixed then. I probably don't want to make a new release with OpsGenie and the raw JSON syntax though, but I can help with suppling the methods for reading and writing those fields.

Reply:
That should work for arrays and maps.

For responders/visibleTo we need to distinguish between a name or an id. Scroll down a bit here for all options: https://docs.opsgenie.com/docs/alert-api#create-alert. It's not clear to me how to do that with your example. 

How about this adding the name/id information to the key like this?

```
&responders=team-id:4513b7ea-3b91-438f-b7e4-e3e54af9147c,team-name:NOC,schedule-id:XXXX
```

Reply:
Are there other services in OpsGenie that need similar complex parameters?

After thinking about it a bit, I believe that most people won't need to set `responders/visibleTo` as URL parameter. The OpsGenie UI/API offer much more sophisticated ways of routing alerts to teams/users. Compared to setting URL parameters, this lets users configure the routing rules centrally, instead of sending them in via every alert.

Still, no harm to get this to work right but using this is probably special case rather then the norm.

Reply:
My example just presumed that all IDs would be in the GUID-ish format, and if it matched that format it would be interpreted as an ID.

If you for some bizarre reason would have a team with the name `4513b7ea-3b91-438f-b7e4-e3e54af9147c`, then you'd have to specify that one by ID instead. That seems like it should cover all cases. But perhaps the IDs can be in another format?

Reply:
I haven't seen anything other then this format so far in the 6 months that I've used OpsGenie. Having said that, there is always the slight possibility that OpsGenie might change their id format in the future. Unlikely but possible.

If you are ok with that, I can go and implement it using your proposed solution.